File: kotlin/modules/domain/src/main/kotlin/dev/zerosum/example/greeting/GreetingService.kt
package dev.zerosum.example.greeting

import java.time.Clock
import java.time.ZonedDateTime
import javax.inject.Named

interface GreetingService {
    fun greet(): String
}

@Named
class GreetingServiceImpl(
        private val clock: Clock
) : GreetingService {

    override fun greet(): String {
        return when (ZonedDateTime.now(clock).hour) {
            in 0..6 -> "Zzz..."
            in 7..11 -> "Good morning!"
            in 12..17 -> "Good afternoon!"
            in 18..23 -> "Good evening!"
            else -> "Hello, World!!"
        }
    }
}

File: scala/modules/domain/src/main/scala/dev/zerosum/example/greeting/GreetingService.scala
package dev.zerosum.example.greeting

import java.time.{Clock, ZonedDateTime}

class GreetingService(val clock: Clock) {

  def greet: String = {
    ZonedDateTime.now(clock).getHour match {
      case h if (0 to 6).contains(h) => "Zzz..."
      case h if (7 to 11).contains(h) => "Good morning!"
      case h if (12 to 17).contains(h) => "Good afternoon!"
      case h if (18 to 23).contains(h) => "Good evening!"
      case _ => "Hello, World!!"
    }
  }
}
